src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -0,0 +1,176 @@
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.flowframework.constant.CommonValue.META;
+import static org.opensearch.flowframework.constant.CommonValue.NO_SCHEMA_VERSION;
+import static org.opensearch.flowframework.constant.CommonValue.SCHEMA_VERSION_FIELD;
+
+/**
+ * Flow Framework Indices Handler
+ */
+public class FlowFrameworkIndicesHandler {
+
+    private static final Logger logger = LogManager.getLogger(FlowFrameworkIndicesHandler.class);
+    private static final Map<String, Object> indexSettings = Map.of("index.auto_expand_replicas", "0-1");
+    private static final Map<String, AtomicBoolean> indexMappingUpdated = new HashMap<>();
+
+    private ClusterService clusterService;
+    private Client client;
+
+    /**
+     * Handler constructor
+     * @param clusterService cluster service
+     * @param client client
+     */
+    public FlowFrameworkIndicesHandler(ClusterService clusterService, Client client) {
+        this.clusterService = clusterService;
+        this.client = client;
+    }
+
+    static {
+        for (FlowFrameworkIndex flowFrameworkIndex : FlowFrameworkIndex.values()) {
+            indexMappingUpdated.put(flowFrameworkIndex.getIndexName(), new AtomicBoolean(false));
+        }
+    }
+
+    /**
+     * Initiate global context index if it's absent
+     * @param listener action listner
+     */
+    public void initGlobalContextIndexIfAbsent(ActionListener<Boolean> listener) {
+        initFlowFrameworkIndexIfAbsent(FlowFrameworkIndex.GLOBAL_CONTEXT, listener);
+    }
+
+    /**
+     * General method for initiate flow framework indices or update index mapping if it's needed
+     * @param index flow framework index
+     * @param listener action listener
+     */
+    public void initFlowFrameworkIndexIfAbsent(FlowFrameworkIndex index, ActionListener<Boolean> listener) {
+        String indexName = index.getIndexName();
+        String mapping = index.getMapping();
+
+        try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+            ActionListener<Boolean> internalListener = ActionListener.runBefore(listener, () -> threadContext.restore());
+            if (!clusterService.state().metadata().hasIndex(indexName)) {
+                ActionListener<CreateIndexResponse> actionListener = ActionListener.wrap(r -> {
+                    if (r.isAcknowledged()) {
+                        logger.info("create index:{}", indexName);
+                        internalListener.onResponse(true);
+                    } else {
+                        internalListener.onResponse(false);
+                    }
+                }, e -> {
+                    logger.error("Failed to create index " + indexName, e);
+                    internalListener.onFailure(e);
+                });
+                CreateIndexRequest request = new CreateIndexRequest(indexName).mapping(mapping).settings(indexSettings);
```

Comment:
does the mapping here require a mediatype?  Looking at the javadoc it seems it expects JSON but needs a special format with a `_doc` key.   We shold be consistent between our various implementations so all our mapping files look consistent.